### PR TITLE
[DOCS] Add example for `TopPLogitsWarper` 

### PR DESCRIPTION
**Pull request by chiral-carbon**
# What does this PR do?

Adds an example to the docstring of `TopPLogitsWarper` class definition in [this file](https://github.com/huggingface/transformers/blob/main/src/transformers/generation/logits_process.py).

Fixes one of the cases in #24783 

## Before submitting
- [x] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [x] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#start-contributing-pull-requests),
      Pull Request section?
- [x] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.

## Who can review?
@gante

### CODE REVIEW

**Review by gante — COMMENTED** [review]
In general LGTM, but let's try to save some lines for a nicer display in the docs :)

E.g.:
- Most of the comments saying what the following lines will do (like `>>> # Import required modules`) can be removed
- `>>> model_name = "gpt2"` -> this doesn't need to be a variable
- we don't need to set any pad token, as we are not adding padding
- make sure you don't have two empty lines in a row

**Comment by chiral-carbon** [comment]
Thanks for the review!

Quick question:

> * we don't need to set any pad token, as we are not adding padding

that is correct, but without the two lines for padding, I get this response when I run `model.generate()` in my own python terminal:
```
The attention mask and the pad token id were not set. As a consequence, you may observe unexpected behavior. Please pass your input's `attention_mask` to obtain reliable results.
Setting `pad_token_id` to `eos_token_id`:50256 for open-end generation.
```
If this is fine, then I will go ahead and remove code for setting the pad token.
If this is

**Comment by gante** [comment]
@chiral-carbon yes, that is fine to illustrate the example :)

**Review by gante — APPROVED** [review]
Awesome, thank you for iterating 👍 I've added two minor nits :)

**Comment by chiral-carbon** [comment]
Thanks @gante, I have addressed the nits. I re-ran the commands in terminal for good measure so you will see different outputs from the previous commit.

**Review by sgugger — APPROVED** [review]
Thanks for adding this!
